### PR TITLE
MCR-2431 improve xml access system

### PR DESCRIPTION
**Pull request by kkrebs**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2431).

### CODE REVIEW

File: mycore-base/src/main/java/org/mycore/access/MCRAccessManager.java [hunk 1]
```diff
@@ -65,11 +65,20 @@
...
-        return (T) MCRConfiguration2.getSingleInstanceOf("MCR.Access.Class", MCRAccessBaseImpl.class).get();
+        return (T) MCRConfiguration2.<MCRAccessInterface>getInstanceOf("MCR.Access.Class")
+            .orElseGet(MCRAccessBaseImpl::new);
     }
 
     private static MCRAccessCheckStrategy getAccessStrategy() {
+        // if acccessStrategy equals accessImp we reuse the accessImpl, 
+        // to make sure, that only one singleton gets created
+        // (used to instantiate fact-based access system) 
+        if (MCRConfiguration2.getString("MCR.Access.Strategy.Class").orElse("UNDEFINED")
```
Comment (sebhofmann) [inline mycore-base/src/main/java/org/mycore/access/MCRAccessManager.java]: This doesnt work if the Strategy is defined in MCR.Access.Strategy (which is the preferred way).

Comment (sebhofmann) [reply mycore-base/src/main/java/org/mycore/access/MCRAccessManager.java]: Also "UNDEFINED" makes no sense since you can check if the optional is null.

**Review by sebhofmann — APPROVED** [review]

**Review by sebhofmann — CHANGES_REQUESTED** [review]
check comments

File: mycore-base/src/main/java/org/mycore/access/MCRAccessManager.java (hunk 1 above)
Comment (rsteph-de) [reply mycore-base/src/main/java/org/mycore/access/MCRAccessManager.java]: reimplemented property comparison

**Review by sebhofmann — APPROVED** [review]

**Pull request merged**